s3store/s3store.go

```diff
@@ -547,3 +560,40 @@ func isAwsError(err error, code string) bool {
 	}
 	return false
 }
+
+func (store S3Store) CalcOptimalPartSize(size int64) (optimalPartSize int64, err error) {
+	var errorstrings []string
+
+	switch {
+	// When upload is smaller or equal MinPartSize, we upload in just one part.
+	case size <= store.MinPartSize:
+		optimalPartSize = store.MinPartSize
+	// When we need 9999 parts or less with MinPartSize.
+	case size/store.MinPartSize < store.MaxMultipartParts:
+		optimalPartSize = store.MinPartSize
+	// If our upload divides up exactly into MaxMultipartParts parts with
+	// no bytes leftover, we will not need an spare last part. So we can go with
+	// a straight division. Otherwise we would exceed MaxPartSize in a scenario,
+	// where MaxObjectSize is equal to MaxPartSize * MaxMultipartParts
+	// (which is not the case with the current AWS S3 API specification, but
+	// might be in the future or with other S3-aware stores).
+	case size%store.MaxMultipartParts == 0:
+		optimalPartSize = size / store.MaxMultipartParts
+	// In all other cases, we need to round up to the next integer.
```

Comment:
Would you be so kind and add to the comment that integer division will round down, if necessary. This will help understanding the "+1"

Reply:
No problem.

Reply:
Doing the division in float64 and then rounding up the result is no option I guess, as we might end up with rounding errors due to the float nature of things?!

It would enhance readability, but probably introduce a whole new set of problems.

Reply:
I believe the errors which occur due to the use of a floating point number is too minimal to be harmful for our usecase but I am not keen on testing that :)
Also, rounding functions in Go also returns a float64 not integers. Let's keep the current approach.